### DISCUSSION TITLE
hysteria2和6.12内核的兼容性

### DISCUSSION THREAD

**Discussion by yyysuo** [apernet/hysteria#1282] [Q&A] [answered]
6.12内核的op当主路由，家里用sing-box跑hy2入站回家，回家的速度只有1Mbps了，lean说需要等hy2更新适配新内核。

**Comment by haruue** [comment]
你用的是 hy2 还是 tcp-brutal ？

**Reply by yyysuo** [reply]
> 你用的是 hy2 还是 tcp-brutal ？

没用tcp-brutal，就是简单用sing-box开了个入站，试了lean大佬的qwrt当主路由和sbwml大佬的x64固件当主路由，升级到6.12.3和6.12.5内核，都会出现这个问题，在6.6X内核的OP当主路由的时候没问题。

    {
       "type": "hysteria2",
       "tag": "hy2",
       "tcp_fast_open": true,
       "listen": "::",
       "listen_port": 777,
       "users": [
              {
              "password": "mypass"
              }
          ],
       "ignore_client_bandwidth": true,
        "tls": {
              "enabled": true,
              "alpn": [
                  "h3"
              ],
              "certificate_path": "/root/hysteria/cert.pem",
              "key_path": "/root/hysteria/private.key"
          }
    },

![image](https://github.com/user-attachments/assets/a0ffedf7-39fc-4daf-8f13-e79a25bf237b)

**Reply by haruue** [reply]
我在 Arch Linux 上用 6.12 内核测试一切正常。
![image](https://github.com/user-attachments/assets/d213f01d-0734-450a-bd09-8d399fec21b4)

> lean说需要等hy2更新适配新内核

所以能不能直接让 lean 来告诉我们应该进行哪些适配。 OpenWrt 的测试环境实在不好弄。

**Reply by yyysuo** [reply]
> 我在 Arch Linux 上用 6.12 内核测试一切正常。 ![image](https://private-user-images.githubusercontent.com/13131972/398169527-d213f01d-0734-450a-bd09-8d399fec21b4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzUwMDA2NTIsIm5iZiI6MTczNTAwMDM1MiwicGF0aCI6Ii8xMzEzMTk3Mi8zOTgxNjk1MjctZDIxM2YwMWQtMDczNC00NTBhLWJkMDktOGQzOTlmZWMyMWI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjI0VDAwMzIzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg0MTJhMjVmZTc4MDg4OTZmYzIwYTE0MzU4YmY1ZmE2ZjhhYzIxODY2NWE0MTVhMGZmYTg1OTRmMTU0NjVhNzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.XHdsl05eXioS3wwP3GAxmiPKgj_KJZsMhud-CZSlmW0)
> 
> > lean说需要等hy2更新适配新内核
> 
> 所以能不能直接让 lean 来告诉我们应该进行哪些适配。 OpenWrt 的测试环境实在不好弄。

辛苦大佬验证，lean大佬之前原话是“6.12的tcp模块变了，hy2还没更新”，具体没细说，我又私信他问了，不过虽然v了50，他也不一定有时间有心情搭理我，之所以发讨论区，也是因为不确定是哪方面的问题，没有进一步细节之前，大佬可以无视这个讨论。

**Reply by yyysuo** [reply, marked as answer]
> 我在 Arch Linux 上用 6.12 内核测试一切正常。 ![image](https://private-user-images.githubusercontent.com/13131972/398169527-d213f01d-0734-450a-bd09-8d399fec21b4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk3Nzg2ODYsIm5iZiI6MTczOTc3ODM4NiwicGF0aCI6Ii8xMzEzMTk3Mi8zOTgxNjk1MjctZDIxM2YwMWQtMDczNC00NTBhLWJkMDktOGQzOTlmZWMyMWI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjE3VDA3NDYyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMyZTk2ZTEwZDhjMjQ1MGFmOWE3YmZiYjc3MTQ2ZDk1OWMxY2E5YTg3NzJmYTM1YzEzMzcyY2E1MzdhMWJhMzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.o-Arczaucd7DzaOLWp0FSm_VXjiy95-kJ8DL9m9Lv9w)
> 
> > lean说需要等hy2更新适配新内核
> 
> 所以能不能直接让 lean 来告诉我们应该进行哪些适配。 OpenWrt 的测试环境实在不好弄。

非常抱歉给大佬造成了困扰，问题搞清楚了，在sing-box启动时禁用gso就可以了。
QUIC_GO_DISABLE_GSO=true /cus/bin/sing-box run -c /cus/singbox/config.json
